### PR TITLE
Peering connection support

### PR DESCRIPTION
**Pull request by blakeneyops**
Added module for establishing a peering connection along with full UAT testing. Also some fixes for  VPG and additional automated testing.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -2,12 +2,35 @@ machine:
   services:
     - docker
   environment:
-    TF_IMG: unifio/terraform:0.7.4
-    TF_CMD: docker run -v /home/ubuntu/.aws:/home/user/.aws -e AWS_DEFAULT_REGION=us-east-1 -e LOCAL_USER_ID=1000 --rm
```
Comment (kjonick1) [inline circle.yml:6]: Should we keep the rm to remove container after running?  I recall containers would eventually fill filesystem.

**Review by kjonick1 — COMMENTED** [review]
Other than docker command, looks good to me.

**Comment by kjonick1** [comment]
Other than docker command, looks good to me.

File: circle.yml (hunk 1 above)
Comment (blakeneyops) [reply circle.yml:6]: Typically, yes. However, this causes a lot of warnings in the Circle environment, which is technically docker-in-docker, so I pulled it out.

Comment (kjonick1) [reply circle.yml:6]: 👍

**Pull request merged**